geodiff/src/drivers/sqlitedriver.cpp

```diff
@@ -533,7 +542,7 @@ static void handleUpdated( const Context *context, const std::string &tableName,
         if ( tbl.columns[i].type == TableColumnType::DATETIME )
         {
           Sqlite3Stmt stmtDatetime;
-          stmtDatetime.prepare( db, "SELECT datetime(?1) IS NOT datetime(?2)" );
+          stmtDatetime.prepare( db, "SELECT STRFTIME('%%Y-%%m-%%d %%H:%%M:%%S.%%fZ', ?1) != STRFTIME('%%Y-%%m-%%d %%H:%%M:%%S.%%fZ', ?2)" );
```

Comment:
A couple of things:
- using `!=` instead of `IS NOT` is going to break support for NULL values (as a result I think you had to modify `geodiff/tests/testdata/datetime/datetime1-2.diff` which should not have been necessary)
- `%f` already includes seconds: `select strftime('%Y-%m-%d %H:%M:%S.%fZ', 'now');` returns seconds twice: `2022-09-02 09:57:09.09.119Z`
- no need to add the extra `Z` for comparison on both sides?

Reply:
Fixed.